value/src/org/immutables/value/Value.java

```diff
@@ -1149,6 +1149,93 @@
     String redactedMask() default "";
 
     /**
+     * When enabled: immutable attributes with discoverable builders will receive the additional
+     * builder API:
+     *
+     * For single children:
+     * {@code get*Builder()}
+     * {@code set*Builder(BuilderT builder)}
+     *
+     * For collection of children:
+     * {@code add*Builder()}
+     * {@code addAll*Builder(Iterable<BuilderT> builderCollection)}
+     * {@code addAll*Builder(BuilderT... builderArgs)}
+     * {@code get*BuilderList()}
+     *
+     * In strict mode, you may only call set*Builder once, but you may call get*Builder multiple
+     * times, in which the same builder is returned. If the nested immutable is also strict,
+     * then you will only be able to set properties on the child builder once.
+     *
+     * To discover builders on value attributes the value methods are scanned for method names
+     * matching a patterns specified in {@link #nestedBuilderDetectionPattern()}.
+     *
+     * This style parameter is experimental and may change in future.
+     * @return true to enable the feature.
+     */
+    boolean nestedBuilderDetection() default false;
+
+    /**
+     * Pattern for discovering builders.
+     *
+     * {@link #nestedBuilderDetectionPattern()} applies to both
+     * static and instance methods. In the case a builder is only discoverable through a
+     * value instance method, the builder class must have a public no arg constructor,
+     * or a no-arg static method constructor.  The static method will be preferred.
+     *
+     * <em>This is detection pattern, not formatting pattern. It defines how to recognize a nested builder.</em>
+     * Only applies if {@link #nestedBuilderDetection()} is {@code true}.
+     *
+     * @return naming template
+     */
+    String[] nestedBuilderDetectionPattern() default {"*Builder", "*builder"};
+
+    /**
+     * Naming template for retrieving a nested builder.
+     *
+     * Only applies if {@link #nestedBuilderDetection()} is {@code true}.
+     *
+     * @return naming template.
+     */
+    String getBuilder() default "get*Builder";
+
+    /**
+     * Naming template for setting a nested builder.
+     * This may be called only once in strict mode.
+     *
+     * Only applies if {@link #nestedBuilderDetection()} is {@code true}.
+     *
+     * @return naming template.
+     */
+    String setBuilder() default "set*Builder";
```

Comment:
I would prefer `default *Builder"` here: by default, for attribute initializers in builder we do not use `set` prefixes, I believe there should be no conflict with `getBuilder` pattern because of the parameter type differs (properly overloaded method)